File: SuperSimpleStockMarket_Java/src/com/sssm/util/Utilities.java
package com.sssm.util;

/**
 * @author Lucky (LakshmiNarayana Ambarkar)
 * 
 * Utility class containing static methods
 */
public class Utilities {
	/**
	 * Logic obtained from https://stackoverflow.com/questions/26839890/calculating-nth-root-of-a-positive-integer-in-java
	 * 
	 * @param n		=> nth root 
	 * @param value => value on which the root has to be calculated
	 * @return		=> nth root of value
	 */
	public static double nThRootOf(int n, double value) {
		if (n==0) {
			return 1;
		}
		
		return Math.pow(value, (1.0/n));
	}
}

File: SuperSimpleStockMarket_Python/com/sssm/util/utilities.py
class MathUtil(object):
    """ Utility method for mathematical calculations that are not available in the standard math library.
        @author Lucky (LakshmiNarayana Ambarkar)
    """
    
    @staticmethod
    def n_root_of_x(n, x):
        """ Calculates the Nth root of a given number.
            Logic obtained from https://stackoverflow.com/questions/26839890/calculating-nth-root-of-a-positive-integer-in-java
        """
        if n==0:
            return 1
        
        return 1 if n==0 else x**(1.0/n)
